compiler/rustc_lint/src/builtin.rs

```diff
@@ -745,6 +750,32 @@ impl<'tcx> LateLintPass<'tcx> for MissingCopyImplementations {
     }
 }
 
+/// Check whether a `ty` has a negative `Copy` implementation, ignoring outlives constraints.
+fn type_implements_negative_copy_modulo_regions<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    ty: Ty<'tcx>,
+    param_env: ty::ParamEnv<'tcx>,
+) -> bool {
+    let trait_ref = ty::TraitRef::new(tcx, tcx.require_lang_item(hir::LangItem::Copy, None), [ty]);
+    let pred = ty::TraitPredicate {
+        trait_ref,
+        constness: ty::BoundConstness::NotConst,
+        polarity: ty::ImplPolarity::Negative,
+    };
+    let obligation = traits::Obligation {
+        cause: traits::ObligationCause::dummy(),
+        param_env,
+        recursion_depth: 0,
+        predicate: ty::Binder::dummy(pred).to_predicate(tcx),
+    };
+
+    tcx.infer_ctxt()
+        .build()
+        .evaluate_obligation(&obligation)
```

Comment:
Can we use [`predicate_may_hold`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_trait_selection/infer/struct.InferCtxt.html#method.predicate_may_hold) here instead?

Reply:
Now using `predicate_may_hold`. That's much shorter and legible, thanks for the suggestion!
However, I'm wondering if that could yield too many false positives since we're now considering *ambiguous* negative impls to apply (as well as ones that apply modulo opaque types).
Wouldn't `predicate_must_hold_modulo_regions` be a better fit?

Reply:
Yes, sorry about that. I think you're correct, `predicate_must_hold_modulo_regions` is the right method to use here I think.

Reply:
Done